### ISSUE TITLE
Unable to build with libjpeg-turbo

### ISSUE THREAD

**Issue by george-geng** [LibVNC/libvncserver#322]
Hi! I've been trying unsuccessfully to build `libvnc` with `Zlib` and `libjpeg-turbo` (in order to use `tight` encoding with my client and server). I've been closely following this more advanced build example from the docs: https://github.com/LibVNC/libvncserver/blob/master/.appveyor.yml

My directory structure for libjpeg-turbo is the same as LibPNG. I've tried linking `turbojpeg.lib`, `turbojpeg-static.lib`, and even `jpeg.lib` and `jpeg-static.lib` to no avail.  While I'm able to link `LibPNG` and `Zlib` by following that example, I keep getting this warning:

```
*** The libjpeg library you are building against is not libjpeg-turbo.
Performance will be reduced. You can obtain libjpeg-turbo from:
https://sourceforge.net/projects/libjpeg-turbo/files/ ***
```

I went to the source forge link and tried 2 different libjpeg turbo versions, 2.0.2 and 1.5.1, but still get the same issue. I've had no issues building either of those libjpeg libraries.

I've also tried building and linking through the cmake-gui as well.

What's more, even if I ignore the warning and go ahead and try to build the `ppmtest.c` client example, I get warnings such as `unresolved external symbol jpeg_write_scanlines`, which makes me wonder if it's even using the ordinary (non-tubo) `libjpeg` library at all.

Any help on this problem would be really appreciated, thank you!

**Comment by george-geng** [comment]
@bk138 I've just found that I'm able to link `libjpeg-turbo` if I use the 32-bit Visual Studio compiler with cmake to build `libvncserver`, but this issue still exists with  64-bit MSVC or MinGW.

**Comment by george-geng** [comment]
@bk138 This issue occurs only when I try to use cmake to create MinGW makefiles. (I was able to get it to work for Win64 and Win32). The reason for me trying MinGW was because I was running into bugs like [this one](https://github.com/LibVNC/libvncserver/issues/165#issuecomment-286619278) with the MSVC builds, but some comments such as [this](https://github.com/LibVNC/libvncserver/issues/165#issuecomment-326248510) or [this] (https://github.com/LibVNC/libvncserver/pull/257#issuecomment-510338950) seemed to suggest that MinGW builds were more stable.

I've tried linking against Win64, Win32, and MinGW builds of libjpeg-turbo but am still getting that error. Is there a solution to this, or is my build process incorrect?

I'm essentially running: ```cmake -G"MinGW Makefiles" .. link to libraries here...```
